apps/api/src/persistence/database/arango-database.ts

```diff
@@ -52,10 +51,23 @@ export class ArangoDatabase {
             throw new InternalError(`Arango cannot fetchById with invalid id: ${id}`);
         }
 
-        // TODO optimize this! Do the fetch by ID in actual AQL
```

Comment:
Note that this was optimized, but involved a difficult to reason about abstraction using the "specification" pattern.